include/RAJA/pattern/params/reducer.hpp

```diff
@@ -111,38 +98,25 @@ namespace detail
 } // namespace detail
 
 template <template <typename, typename, typename> class Op, typename T>
-auto constexpr Reduce(T *target)
+auto constexpr Reduce(ValOp<T,Op> *target)
 {
-  return detail::Reducer<Op<T, T, T>, T>(target);
+  return detail::Reducer<Op, T, ValOp<T,Op>>(target);
 }
 
+//template <template <typename, typename, typename> class Op, typename T, typename IndexType,
+//           std::enable_if_t<std::is_integral<T>::value || std::is_floating_point<T>::value>* = nullptr >
+//auto constexpr Reduce(ValLoc<T, IndexType> *target)
+//{
+//  return detail::Reducer<Op, T, ValLoc<T, IndexType>>(target);
+//}
 
+//template <template <typename, typename, typename> class Op, typename T,
+//           std::enable_if_t<std::is_integral<T>::value || std::is_floating_point<T>::value>* = nullptr >
+//auto constexpr Reduce(T *target)
+//{
+//  return detail::Reducer<Op, T, T>(target);
+//}
```

Comment:
no longer used?

Reply:
I'm recommending not supporting the old interface of passing in basic data types, and requiring the user to pass in a `ValOp`. As such, these overloads will not be necessary.

Reply:
We will need to make sure the change is documented well in our user guide with good before/after examples. Do you plan to add that to this PR?

Reply:
Indeed, I've added Documentation in the Todo list in this PR's description up top. I'll include a table of before & after code in the description of this PR, after I get the tests sorted out properly.